// @yunyoujun/logger/dist/logger.js
"use strict";
/**
 * 辅助工具，输出彩色控制台信息。
 * @packageDocumentation
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const chalk_1 = __importDefault(require("chalk"));
const options_1 = require("./options");
/**
 * 日志工具类
 */
class Logger {
    options;
    /**
     * 启用
     */
    enable = true;
    /**
     * 当前模式，开发模式输出 Debug 信息
     */
    mode = "dev";
    constructor(options) {
        this.options = options;
        this.options = Object.assign(options_1.defaultOptions, options);
    }
    /**
     * 打印消息
     * @param type 类型
     * @param msg 消息
     */
    print(type, msg) {
        if (!this.enable)
            return;
        const color = {
            success: "green",
            warning: "yellow",
            error: "red",
            info: "blue",
            debug: "cyan",
        };
        const typeColor = color[type];
        const typeName = `[${type}]`;
        const content = [];
        if (this.options.prefix) {
            content.push(this.options.prefix);
        }
        if (this.options.type) {
            content.push(chalk_1.default[typeColor](typeName));
            content.push(msg);
        }
        else {
            content.push(chalk_1.default[typeColor](msg));
        }
        console.log(...content);
    }
    /**
     * 输出成功信息（绿色）
     * @param msg 文本
     */
    success(msg) {
        this.print("success", msg);
    }
    /**
     * 输出警告信息（黄色）
     * @param msg 文本
     */
    warning(msg) {
        this.print("warning", msg);
    }
    /**
     * 输出错误信息（红色）
     * @param msg 文本
     */
    error(msg) {
        this.print("error", msg);
    }
    /**
     * 输出提示信息（蓝色）
     * @param msg 文本
     */
    info(msg) {
        this.print("info", msg);
    }
    /**
     * 输出 Debug 信息（cyan）
     * @param msg
     */
    debug(msg) {
        if (this.mode === "dev") {
            this.print("debug", msg);
        }
    }
}
exports.default = Logger;
